receiver/oracledbreceiver/scraper.go

```diff
@@ -828,3 +828,43 @@ func (s *oracleScraper) shutdown(_ context.Context) error {
 	}
 	return s.db.Close()
 }
+
+func (s *oracleScraper) setupResourceBuilder(rb *metadata.ResourceBuilder) *metadata.ResourceBuilder {
+	rb.SetOracledbInstanceName(s.instanceName)
+	rb.SetHostName(s.hostName)
+	rb.SetServiceInstanceID(s.serviceInstanceID)
+	return rb
+}
+
+func getInstanceID(instanceString string, logger *zap.Logger) string {
+	parts := strings.SplitN(instanceString, "/", 2)
+	hostAndPort := parts[0]
+	service := ""
+	if len(parts) == 2 {
+		service = "/" + parts[1]
```

Comment:
I think the value of `service` would be a lot easier to follow if we just made it be the service itself, and then use `fmt.Sprintf` later to add the `/`. The `handleInstanceIDErr` method is assuming the format of `service` starts with `/`, which may not be obvious to another user who may call this method down the road. The returned `service.instance.id` would then no longer match the expected format.